pkg/util/kubernetes/apiserver/store.go

```diff
@@ -93,3 +94,20 @@ func (m *metaBundleStore) delete(nodeName string) {
 
 	m.cache.Delete(cacheKey)
 }
+
+// SkipKubernetesResource checks with a global kubernetes cache whether the resource was already reported.
+// It will return true in case the UID is in the cache and the resourceVersion did not change. Else it will return false.
+// 0 == defaultDuration
+func SkipKubernetesResource(uid types.UID, resourceVersion string) bool {
+	cacheKey := agentcache.BuildAgentKey("kubernetes", "resource", string(uid))
+	value, hit := agentcache.Cache.Get(cacheKey)
+	if !hit {
+		agentcache.Cache.Set(cacheKey, resourceVersion, 0)
```

Comment:
0 indicates defaultExpiration time, which we default to 5 minutes. We can make it alternatively configurable in this PR

Reply:
could be useful to make it configurable

Reply:
ah sorry I didn't know you already discussed that already 😅  I'm just assuming this function could be used by other components given that it's placed in this package, not in the orchestrator package